src/Npgsql/PostgresException.cs

```diff
@@ -28,6 +28,15 @@ public sealed class PostgresException : NpgsqlException
     {
         bool _dataInitialized;
 
+        /// <summary>
+        /// Creates a new instance.
+        /// </summary>
+        /// <remarks>
+        /// Exists for backwards compat with 4.0, has been removed for 5.0.
+        /// </remarks>
+        [Obsolete]
+        public PostgresException() : this(string.Empty, string.Empty, string.Empty, string.Empty) {}
```

Comment:
Should we be marking obsolete other things tagged "Exists for backwards compat with 4.0, has been removed for 5.0."?

Reply:
Unless I'm mistaken, most/all other constructors are there only to prevent a binary breaking change. For example, instead of `ExecuteReader(CommandBehavior behavior = default)`, I added back `ExecuteReader()` alongside `ExecuteReader(CommandBehavior behavior)`. This means the exact same code will compile just fine against 5.0 - it's just that binaries built against 4.x won't run against 5.0. We don't need to put [Obsolete] on those.

Reply:
Was thinking about the `ctor` for `FieldDescription` and relocation of `NpgsqlTimeout`, but was just skimming so might be a non-issue.

Reply:
Yeah, the FieldDescription ctor does make sense, although it's really a super-internal thing to begin with anyway. It's only public so that we can do type handlers, since these accept FieldDescription in their APIs. There's almost zero chance anyone is using this - but go ahead and add it if you want.

NpgsqlTimeout is just a relocation, so again I don't think it's right to obsolete it... In 5.0 IDEs will automatically suggest the new namespace etc.